sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -212,8 +213,14 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
         f.fallbackFileFormat
       case _ => lookupCls
     }
+    val clsMustUseV1Source = cls.newInstance() match {
+      case f: DataSourceRegister if useV1Sources.contains(f.shortName()) ||
```

Comment:
cc @gengliangwang  the `useV1SourceReaderList` should be applied to all data sources not only file sources, according to the doc. Can you fix it? The current patch works but it's not a clean solution.

Reply:
Yes, I  think we can update the code in line 210 directly.

Reply:
Thanks for the fix! Just ping me and I can review it right away.

Reply:
Created https://github.com/apache/spark/pull/25004

Reply:
Thanks for the help!